src/wallet.ts

```diff
@@ -0,0 +1,287 @@
+    // account lookups
+    'eth_accounts': createAsyncMiddleware(lookupAccounts),
+    'eth_coinbase': createAsyncMiddleware(lookupDefaultAccount),
+    // tx signatures
+    'eth_sendTransaction': createAsyncMiddleware(sendTransaction),
+    // message signatures
+    'eth_sign': createAsyncMiddleware(ethSign),
+    'eth_signTypedData': createAsyncMiddleware(signTypedData),
+    'eth_signTypedData_v3': createAsyncMiddleware(signTypedDataV3),
+    'eth_signTypedData_v4': createAsyncMiddleware(signTypedDataV4),
+    'personal_sign': createAsyncMiddleware(personalSign),
+    'eth_getEncryptionPublicKey': createAsyncMiddleware(encryptionPublicKey),
+    'eth_decrypt': createAsyncMiddleware(decryptMessage),
+    'personal_ecRecover': createAsyncMiddleware(personalRecover),
+  });
+
+  //
+  // account lookups
+  //
+
+  async function lookupAccounts(req: JsonRpcRequest<unknown>, res: PendingJsonRpcResponse<unknown>): Promise<void> {
+    res.result = await getAccounts(req);
+  }
+
+  async function lookupDefaultAccount(req: JsonRpcRequest<unknown>, res: PendingJsonRpcResponse<unknown>): Promise<void> {
+    const accounts = await getAccounts(req);
+    res.result = accounts[0] || null;
+  }
+
+  //
+  // transaction signatures
+  //
+
+  async function sendTransaction(req: JsonRpcRequest<unknown>, res: PendingJsonRpcResponse<unknown>): Promise<void> {
+
+    if (!processTransaction) {
+      throw ethErrors.rpc.methodNotSupported({});
+    }
+
+    const txParams: ParamsObject = (req.params as ParamsObjectList)[0] || {};
+    txParams.from = await validateAndNormalizeKeyholder((txParams.from as string), req);
+    res.result = await processTransaction(txParams, req);
+  }
+
+  //
+  // message signatures
+  //
+
+  async function ethSign(req: JsonRpcRequest<unknown>, res: PendingJsonRpcResponse<unknown>): Promise<void> {
+
+    if (!processEthSignMessage) {
+      throw ethErrors.rpc.methodNotSupported({});
+    }
+
+    const address: string = await validateAndNormalizeKeyholder((req.params as string[])[0], req);
+    const message: string = (req.params as string[])[1];
+    const extraParams: ParamsObject = (req.params as ParamsObjectList)[2] || {};
+    const msgParams: ParamsObject = Object.assign({}, extraParams, {
+      from: address,
+      data: message,
+    });
+
+    res.result = await processEthSignMessage(msgParams, req);
+  }
+
+  async function signTypedData(req: JsonRpcRequest<unknown>, res: PendingJsonRpcResponse<unknown>): Promise<void> {
+
+    if (!processTypedMessage) {
+      throw ethErrors.rpc.methodNotSupported({});
+    }
+
+    const message: string = (req.params as string[])[0];
+    const address: string = await validateAndNormalizeKeyholder((req.params as string[])[1], req);
+    const version = 'V1';
+    const extraParams: ParamsObject = (req.params as ParamsObjectList)[2] || {};
+    const msgParams: ParamsObject = Object.assign({}, extraParams, {
+      from: address,
+      data: message,
+    });
```

Comment:
For stylistic purposes (and arguably readability), it would be nice to convert the `Object.assign` calls in this file to object spreads, like so:
```suggestion
    const msgParams: ParamsObject = {
      ...extraParams,
      from: address,
      data: message,
    });
```